srcpkgs/libgccjit/template

```diff
@@ -0,0 +1,241 @@
+	fi
+
+	case "$XBPS_TARGET_MACHINE" in
+		ppc*) _args+=" --disable-vtable-verify";;
+		*) _args+=" --enable-vtable-verify";;
+	esac
+
+	export CFLAGS="${CFLAGS/-D_FORTIFY_SOURCE=2/}"
+	export CXXFLAGS="${CXXFLAGS/-D_FORTIFY_SOURCE=2/}"
+
+	# Disable explicit -fno-PIE, gcc will figure this out itself.
+	export CFLAGS="${CFLAGS//-fno-PIE/}"
+	export CXXFLAGS="${CXXFLAGS//-fno-PIE/}"
+	export LDFLAGS="${LDFLAGS//-no-pie/}"
+
+	_args+=" --prefix=/usr"
+	_args+=" --mandir=/usr/share/man"
+	_args+=" --infodir=/usr/share/info"
+    _args+=" --with-bugurl='https://github.com/jashandeep-sohi/libgccjit-pkg/issues'" \
+	_args+=" --libexecdir=/usr/lib${XBPS_TARGET_WORDSIZE}"
+	_args+=" --libdir=/usr/lib${XBPS_TARGET_WORDSIZE}"
+	_args+=" --with-system-zlib"
+	_args+=" --with-linker-hash-style=$_hash"
+	_args+=" --enable-linker-build-id"
+	_args+=" --enable-threads=posix"
+	_args+=" --enable-lto"
+	_args+=" --enable-shared"
+    _args+=" --enable-host-shared"
+	_args+=" --enable-checking=release"
+	_args+=" --enable-plugins"
+	_args+=" --enable-default-ssp"
+	_args+=" --enable-default-pie"
+	_args+=" --disable-multilib"
+	_args+=" --disable-bootstrap"
+	_args+=" --enable-__cxa_atexit"
+    _args+=" --disable-libquadmath"
+    _args+=" --disable-libquadmath-support"
+    _args+=" --disable-libgomp" 
+    _args+=" --disable-liboffloadmic" 
+    _args+=" --disable-libada"
+    _args+=" --disable-libvtv"
+    _args+=" --disable-libsanitizer"
+	_args+=" --disable-werror"
+	_args+=" --disable-nls"
+	_args+=" --disable-libstdcxx-pch"
+	_args+=" --disable-sjlj-exceptions"
+	_args+=" --disable-target-libiberty"
+
+	mkdir -p build
+	cd build
+
+    # If we are checking packages, we need to have c++ compiled
+    [ -z "$XBPS_CHECK_PKGS" ] && _langs="jit" || _langs="jit,c++"
+    _args+=" --enable-languages=${_langs}"
+    CONFIG_SHELL=/bin/bash \
+	    ${wrksrc}/configure ${_args}
+
+}
+
+do_build() {
+	if [ -z "$CHROOT_READY" ]; then
+		export LD_LIBRARY_PATH="${XBPS_MASTERDIR}/usr/lib${XBPS_TARGET_WORDSIZE}"
+	fi
+    cd build
+    make ${makejobs}
+}
+
+post_build() {
+    # Lets create a staging folder to install into
+    mkdir -p "${wrksrc}/install"
+    cd "${wrksrc}/build"
+    make DESTDIR="${wrksrc}/install" -C gcc install
+
+}
+
+do_check() {
+    cd "${wrksrc}/build/gcc"
+    make check-jit RUNTESTFLAGS="-v -v -v"
+}
+
+pre_install() {
+	if [ "$CROSS_BUILD" ]; then
+		# XXX otherwise links to host libpthread
+		for f in $(find ${wrksrc}/build/${XBPS_CROSS_TRIPLET} -type f -name libtool); do
+			sed -e "s,add_dir=\"-L\$libdir\",add_dir=\"-L${XBPS_CROSS_BASE}\$libdir\",g" -i $f
+		done
+	fi
+}
+
+do_install() {
+    _includehostdir="usr/lib/gcc/${_triplet}/${_minorver}/include"
+    
+    # Get the shlibs over
+    cd "${wrksrc}/install"
+
+    vinstall "${wrksrc}/install/usr/lib64/libgccjit.so.0.0.1" 0755 usr/lib
+    ln -sf libgccjit.so.0.0.1 "${DESTDIR}"/usr/lib/libgccjit.so.0
+    ln -sf libgccjit.so.0.0.1 "${DESTDIR}"/usr/lib/libgccjit.so
```

Comment:
Ideally we would let it create  these symlinks itself. No make target available for it?

Reply:
I use this when I stage the install in post_build. weirdly, the make script doesn't create the symlinks, so there has to be manual symlink creation anyways. I chose the more targeted installation because the default behavior is to install the entire gcc toolchain, so then I would have to go ahead and clear out all the unnecessary files. It seems much simpler to install the sonames and headers directly to isolate its libgccjit's contents from conflicting with gcc, and in line with the documentation which specifies that these are the only necessary components to work with libgccjit on top of an existing gcc installation.